src/components/Link/Link.scss

```diff
@@ -49,3 +49,10 @@
 .ms-Link {
   @include ms-Link;
 }
+
+// For button links we need to ensure by default that we inherit the font size. Otherwise
+// the button default font size will override.
+
+BUTTON.ms-Link {
+  font-size: inherit;
```

Comment:
Instead of setting this on `button`, let's just remove `font-size` from the Link styles completely, so that it is inherited by default.